packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -411,9 +411,14 @@ export class FileStorage implements DStore {
               fname: n.fname,
             }
           );
-          cacheUpdates[n.fname].data.links = links;
+          const unreferencedLinks = await LinkUtils.findUnreferencedLinks({
```

Comment:
would be great if we could do this asynchronously and not need to `await` on this.

ideally, we would setup a websocket connection between engine server and the plugin so engine can notify the plugin when its finished long running tasks (eg. indexing all links)
not in scope for this issue. for now, I would put unreferenced links behind a feature flag (eg. `beta.enableUnreferedLinks`)

Reply:
would be good to have profiling information around this. see profile instructions here -> [[Profiling|dendron.dev.qa#profiling]]

Reply:
> for now, I would put unreferenced links behind a feature flag (eg. beta.enableUnreferedLinks)

Should this be a `DendronDevConfig` ?

> would be good to have profiling information around this.

I'll add some profiling around the existing implementation and move on to your suggested async implementation right after I get the unref links to display on the backlink panel.

Reply:
actually, that function didn't even need the `await`. It's a remnant of an older code that I had that I forgot to remove. 😅 

I'll put this behind a flag for now and after the minimum viable feature ships I'll work on fixing it to your suggestion.